### PR TITLE
Fix issue#685

### LINKED ISSUE #685: The BUG of OccupiableBucketLeapArrayTest.testWindowAfterOneInterval()
## Issue Description

Type: *bug report*

### Describe what happened (or what feature you want)

The BUG is occasional.

I wrote the analysis in the following blog:
https://www.yuque.com/zhaixiaoxiang/cwrx8f/zzy18c

### How to reproduce it (as minimally and precisely as possible)

Execute testWindowAfterOneInterval() many times, we will eventually get failure. (may be 5 times is enough)

### PR DESCRIPTION
**Pull request by zhaixiaoxiang**
### Describe what this PR does / why we need it
Fix #685 

### Describe how you did it
Pls see this blog for analysis: https://www.yuque.com/zhaixiaoxiang/cwrx8f/zzy18c

### Describe how to verify it
Execute this test case many times to ensure it, because the BUG would happen by chance.

### CODE REVIEW

**Comment by sczyh30** [comment]
Hi, thanks for contributing. Please make sure the email of your commits matches your GitHub email.

---

感谢贡献，请将 commit 对应的 email 调整成与 GitHub 的 email 相匹配。

File: sentinel-core/src/main/java/com/alibaba/csp/sentinel/slots/statistic/metric/occupy/OccupiableBucketLeapArray.java [hunk 1]
```diff
@@ -98,4 +98,20 @@ public void debug(long time) {
         }
         System.out.println(sb.toString());
     }
+
+    /**
+     * just for test
+     * @param timeMillis
+     * @return
+     */
+    public long currentWaitingJustForTest(long timeMillis) {
```
Comment (sczyh30) [inline sentinel-core/src/main/java/com/alibaba/csp/sentinel/slots/statistic/metric/occupy/OccupiableBucketLeapArray.java]: Adding a public method just for testing is not recommended. Maybe we can refer to #546 to fix the time-based unstable test cases. Use `AbstractTimeBasedTest` to adjust current time.

Comment (zhaixiaoxiang) [reply sentinel-core/src/main/java/com/alibaba/csp/sentinel/slots/statistic/metric/occupy/OccupiableBucketLeapArray.java]: OK, I have solved it.

Comment (sczyh30) [reply sentinel-core/src/main/java/com/alibaba/csp/sentinel/slots/statistic/metric/occupy/OccupiableBucketLeapArray.java]: 👍

**Comment by zhaixiaoxiang** [comment]
@sczyh30 Is there anything else need improve ?

**Comment by sczyh30** [comment]
> @sczyh30 Is there anything else need improve ?

Since the `AbstractTimeBasedTest` will mock `TimeUtil`, for every test cases in this class, we may need to change `long currentTime = TimeUtil.currentTimeMillis()` to:

```java
long currentTime = System.currentTimeMillis();
setCurrentMillis(currentTime);
```

Or the `currentTime` will be 0 by default.

**Comment by zhaixiaoxiang** [comment]
@sczyh30 Thank you for your remind, I've modified.

**Comment by sczyh30** [comment]
Thanks for contributing!

**Pull request merged**